src/cljs/villagebookUI/components/utils.cljs

```diff
@@ -0,0 +1,30 @@
+(ns villagebookUI.components.utils)
+
+(defn loading
+  []
+  [:div.loading-mask
+   [:div.loading-mask-center
+    [:span.large-blue-spinner]]])
+
+(defn- assoc-target-value
+  [attributes event-name]
+  (if-let [event (event-name attributes)]
+    (assoc attributes event-name #(event (-> % .-target .-value)))
+    attributes))
+
+(defn input
+  [attributes]
+  [:input (assoc-target-value attributes :on-change)])
+
+(defn label
```

Comment:
A coloured label and a colour picker input should be two separate components, not two arities on the same component (particularly not on a component named `label`. Having a separate `colour-input` component would be better.

Reply:
I liked the idea of abstracting away the way an organisation label is implemented. If you pass it an onchange handler it allows itself to be changed, rendering an input field. Otherwise it's a simple div. Is it a bad generalisation?